app/src/test/java/com/codesoom/assignment/controllers/ProductControllerWebTest.java

```diff
@@ -0,0 +1,317 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.AutoConfigureUtf8MockMvc;
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.domain.ProductRepository;
+import com.codesoom.assignment.dto.ProductData;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.hamcrest.core.StringContains;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureUtf8MockMvc
+@DisplayName("ProductController MockMVC 테스트에서")
+public class ProductControllerWebTest {
+    private static final String PRODUCT_NAME = "상품1";
+    private static final String PRODUCT_MAKER = "메이커1";
+    private static final Integer PRODUCT_PRICE = 100000;
+    private static final String PRODUCT_IMAGE_URL = "https://img1.daumcdn.net/thumb/R1280x0/?scode=mtistory2&fname=https%3A%2F%2Ft1.daumcdn.net%2Fcfile%2Ftistory%2F9941A1385B99240D2E";
+
+    private static final String UPDATE_PRODUCT_NAME = "상품1000";
+    private static final Integer UPDATE_PRODUCT_PRICE = 100;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    /**
+     * 하나의 Product 를 생성해 등록합니다.
+     * @return 생성한 Product를 리턴
+     */
+    private Product createProduct() {
+        Product product = Product.builder()
+                .name(PRODUCT_NAME)
+                .maker(PRODUCT_MAKER)
+                .price(PRODUCT_PRICE)
+                .imageUrl(PRODUCT_IMAGE_URL)
+                .build();
+        return productRepository.save(product);
+    }
+
+    @BeforeEach
+    void setUp() {
+        productRepository.deleteAll();
+    }
+
+    @Nested
+    @DisplayName("GET - /products 요청시")
```

Comment:
테스트 대상을 명시해주는 것으로 충분합니다.
```suggestion
    @DisplayName("GET - /products")
```

Reply:
변경했습니다!